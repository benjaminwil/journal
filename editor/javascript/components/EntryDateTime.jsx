import React, {Component} from 'react';

const EntryDateTime = ({datetime}) => (
  <time className="created-at" dateTime={datetime.iso.created}>
    {datetime.created}
  </time>
);

export default EntryDateTime;
