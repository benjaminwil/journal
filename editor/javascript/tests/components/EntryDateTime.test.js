import React from 'react';
import renderer from 'react-test-renderer';
import EntryDateTime from '../../components/EntryDateTime.jsx';

test("the EntryDateTime component displays an entry's creation date", () => {
  const props = {
    datetime: {
      created: 'A million years ago',
      iso: {created: '2019-09-02T19:47:41.113Z'},
    },
  };

  const component = renderer
    .create(<EntryDateTime datetime={props.datetime} />)
    .toJSON();

  expect(component).toMatchSnapshot();
});
