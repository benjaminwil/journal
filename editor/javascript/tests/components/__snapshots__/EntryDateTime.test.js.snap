// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the EntryDateTime component displays an entry's creation date 1`] = `
<time
  className="created-at"
  dateTime="2019-09-02T19:47:41.113Z"
>
  A million years ago
</time>
`;
